schematic/utils/df_utils.py

```diff
@@ -15,26 +19,31 @@ def load_df(
     file_path: str,
     preserve_raw_input: bool = True,
     data_model: bool = False,
-    **load_args,
+    **load_args: dict,
 ) -> pd.DataFrame:
     """
     Universal function to load CSVs and return DataFrames
     Parses string entries to convert as appropriate to type int, float, and pandas timestamp
-    Pandarallel is used for type inference for large manfiests to improve performance
+    Pandarallel is used for type inference for large manifests to improve performance
+
     Args:
-        file_path: path of csv to open
-        preserve_raw_input: Bool. If false, convert cell datatypes to an inferred type
-        data_model: bool, indicates if importing a data model
-        load_args: dict of key value pairs to be passed to the pd.read_csv function
-        **kwargs: keyword arguments for pd.read_csv()
+        file_path (str): path of csv to open
+        preserve_raw_input (bool, optional): If false, convert cell datatypes to an inferred type
+        data_model (bool, optional): bool, indicates if importing a data model
+        **load_args(dict): dict of key value pairs to be passed to the pd.read_csv function
 
-    Returns: a processed dataframe for manifests or unprocessed df for data models and where indicated
+    Returns:
+        pd.DataFrame: a processed dataframe for manifests or unprocessed df for data models and
+      where indicated
     """
     # start performance timer
     t_load_df = perf_counter()
 
     # Read CSV to df as type specified in kwargs
-    org_df = pd.read_csv(file_path, keep_default_na=True, encoding="utf8", **load_args)
+    org_df = pd.read_csv(  # type: ignore
+        file_path, keep_default_na=True, encoding="utf8", **load_args
+    )
+    assert isinstance(org_df, pd.DataFrame)
```

Comment:
That's a really good question. This assert statement is used to tell mypy org_df is a dataframe. This is done because read_csv doesn't always return a dataframe. See the [returns](https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html) section at the bottom. This only happens when the `chunksize` parameter is used, butit's impossible to tell from the fucntion because we use **load_args, which chunksize could be a part of.

Reply:
So to answer your question, we would ahve to figure out if where if its possible to set the `chunksize` parameter upstream. If it's not then we don't need to raise an exception here. If it is possible, than perpahps we should?

Reply:
I went ahead and add the check, its better to be safe! :)

Reply:
That was a really helpful explanation; thank you! Theoretically it would be possible for someone to use the `chunksize` parameter during future development, however I'm not sure if we'd necessarily want to allow that for this specific function. Given that the original intention for this function was to only load a `DataFrame` and the subsequent logic assumes that it's a `DataFrame` rather than a `TextFileReader` I think it makes sense to disallow the latter with an exception like you've done now. I agree, it's better to be safe! Thanks!
@andrewelamb

Reply:
Yeah even if it wasn't possible now ( I have no idea where the load_args come from), this util coudl be used in the future by any number of functions which could expose any/all of the read_csv arguments to a user.

Reply:
Yes, I think there's one place in the codebase that uses that parameter, but I wanted to leave the option to use other parameters open since all calls to `pd.read_csv` in schematic should be going through this function. It's meant to be used during development though, not set by a user when they're running commands